tools/migrators/lib/tests/splunk/resources/test_schedules.py

```diff
@@ -0,0 +1,903 @@
+                                ),
+                                mask3=_generate_splunk_schedule_rotation_shift_mask(
+                                    off_days=["m", "t", "w", "f", "sa", "su"],
+                                    start_hour=0,
+                                    start_minute=0,
+                                    end_hour=9,
+                                    end_minute=0,
+                                ),
+                            ),
+                        ]
+                    ),
+                ],
+                overrides=[],
+                oncall_schedule=_generate_oncall_schedule(id=ONCALL_SCHEDULE_ID),
+            ),
+            DEFAULT_SPLUNK_USERNAME_TO_ONCALL_USER_ID_MAP,
+            ONCALL_SCHEDULE_ID,
+            [
+                _generate_oncall_shift_create_api_payload(
+                    {
+                        "name": "multi day shift1",
+                        "level": 1,
+                        "type": "rolling_users",
+                        "rotation_start": "2024-04-29T13:00:00",
+                        "start": "2024-04-29T13:00:00",
+                        "duration": 60 * 60 * 64,  # 8 hours
+                        "frequency": "weekly",
+                        "interval": 1,
+                        "rolling_users": [[ONCALL_USER1_ID], [ONCALL_USER2_ID]],
+                    }
+                ),
+                _generate_oncall_shift_create_api_payload(
+                    {
+                        "name": "multi day shift2",
+                        "level": 1,
+                        "type": "rolling_users",
+                        "rotation_start": "2024-04-29T13:00:00",
+                        "start": "2024-04-29T13:00:00",
+                        "duration": 60 * 60 * 16,  # 16 hours
+                        "frequency": "weekly",
+                        "interval": 1,
+                        "rolling_users": [[ONCALL_USER1_ID], [ONCALL_USER2_ID]],
+                    }
+                ),
+            ],
+            _generate_oncall_schedule_create_api_payload(_generate_schedule_name(), 2),
+        ),
+    ],
+)
+def test_migrate_schedule(
+    mock_oncall_client,
+    splunk_schedule,
+    user_id_map,
+    expected_oncall_schedule_id_to_be_deleted,
+    expected_oncall_shift_create_calls,
+    expected_oncall_schedule_create_call,
+):
+    schedules.migrate_schedule(splunk_schedule, user_id_map)
+
+    if expected_oncall_schedule_id_to_be_deleted is not None:
+        mock_oncall_client.delete.assert_called_once_with(
+            f"schedules/{expected_oncall_schedule_id_to_be_deleted}"
+        )
+
+    expected_oncall_api_create_calls_args = [
+        ("on_call_shifts", shift) for shift in expected_oncall_shift_create_calls
+    ]
+    expected_oncall_api_create_calls_args.append(
+        ("schedules", expected_oncall_schedule_create_call)
+    )
+
+    print(mock_oncall_client.create.call_args_list)
```

Comment:
I guess we don't need the print here (and below)?

Reply:
good catch, thanks! 👍